username_api.py

```diff
@@ -46,7 +70,11 @@ def check_username(website, username):
 def check_format(website, username):
 	"""Check the format of a username depending on the website"""
 
-	website_parts = patterns['username_patterns'][website]
+	website_parts = patterns.get('username_patterns').get(website, {})
```

Comment:
err, we should always have username patterns for the website ...

here are the facebook published rules : https://www.facebook.com/help/105399436216001#What-are-the-guidelines-around-creating-a-custom-username?

- min: 5 characters (excluding any .)
- permitted characters: A–Z, 0–9, `.`
- not permitted: `.(com|net)`

This is an interesting one, because `.` is permitted and is ignored.  We need a extra pattern in our data for stripping some characters, so that `.` can be stripped from the user input, otherwise the min 5 characters wont be correctly applied.

I am quite sure that this extra pattern will be used by some other websites also.

Facebook also has a fairly common rule that capitalisation can't be used to differentiate usernames.  I think we can assume case-insensitivity as the default for all website, and maybe we need a `case-sensitivity: true` override if we find a website that breaks this rule, and it would be useful to show that the username is available but there is an extremely similar username with a different capitalisation. (Im now on a missing to find a case-sensitive username website ...)

Reply:
Worth noting that there are two parts to case-sensitivity.  One is whether any capitalisation of the username is usable and refers to the same identity.  The other is whether the system allows two usernames with alternative capitalisation.  Wikimedia (MediaWiki is an example of case sensitive usernames , but multiple accounts differing by capitalisation are not permitted)

Reply:
This change isnt needed any more.  If you merge this change, other people wont be forced to add this very important metadata for other websites.